arg-parsing-test.bats

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bats
+
+source lib/assert.bash
+source deploy.sh --source-only
+
+setup() {
+	run mktemp -dt deploy_test.XXXX
+	assert_success
+	tmp=$output
+	pushd "$tmp" >/dev/null
+}
+
+teardown() {
+	popd >/dev/null
+	rm -rf "$tmp"
+}
+
+set_env_vars() {
+	# Set environment variables.
+	export GIT_DEPLOY_USERNAME=env-username
+	export GIT_DEPLOY_EMAIL=env-email
```

Comment:
did you have a plan for these email and username variables? Otherwise, looks great! I'll merge now.